recipes/libinput/all/conanfile.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class LibinputConan(ConanFile):
+    name = "libinput"
+    description = "libinput is a library that handles input devices for display servers and other applications that need to directly deal with input devices."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.freedesktop.org/wiki/Software/libinput/"
+    topics = ("device", "display", "event", "input")
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "epoll_dir": [None, "ANY"],
+        "debug_gui": [True, False],
+        "with_libudev": ["eudev", "systemd"],
+        "with_libwacom": [True, False],
+        "with_wayland": [True, False],
+        "with_x11": [True, False],
+    }
+    default_options = {
+        "epoll_dir": None,
+        "debug_gui": False,
+        "with_libudev": "systemd",
+        # todo Package libwacom and enable this option by default.
+        "with_libwacom": False,
+        "with_wayland": True,
+        "with_x11": True,
+    }
+
+    def configure(self):
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+        if not self.options.debug_gui:
+            self.options.rm_safe("with_wayland")
+            self.options.rm_safe("with_x11")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("mtdev/1.1.6")
+        self.requires("libevdev/1.13.1")
+
+        if self.options.debug_gui:
+            self.requires("cairo/1.18.0")
+            self.requires("glib/2.78.3")
+            self.requires("gtk/system")
+            if self.options.with_wayland:
+                self.requires("wayland/1.22.0")
+                self.requires("wayland-protocols/1.33")
+            if self.options.with_x11:
+                self.requires("xorg/system")
+
+        if self.options.with_libudev == "systemd":
+            # todo Use libudev from the libsystemd package.
+            # self.requires("libsystemd/255.2", transitive_libs=True)
+            self.requires("libudev/system", transitive_libs=True)
+        elif self.options.with_libudev == "eudev":
+            self.requires("eudev/3.2.14", transitive_libs=True)
+
+    def validate(self):
+        if self.settings.os not in ["FreeBSD", "Linux"]:
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported on {self.settings.os}.")
+        if self.options.with_libwacom:
+            raise ConanInvalidConfiguration(f"The with_libwacom option for {self.ref} is not yet supported. Contributions welcome.")
+
+    def build_requirements(self):
+        self.tool_requires("meson/1.3.2")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/2.1.0")
+        if self.options.get_safe("with_wayland"):
```

Comment:
Is wayland needed when `debug_gui` is false?

Reply:
Nope, nice catch!

Reply:
Fixed.